packages/lib/entityPermissionUtils.server.ts

```diff
@@ -1,125 +1,75 @@
+    });
 
-export async function canAccessEntity(
-  entity: Parameters<typeof getEntityPermissionLevel>[0],
-  userId: Parameters<typeof getEntityPermissionLevel>[1]
-) {
-  const permissionLevel = await getEntityPermissionLevel(entity, userId);
-  return (
-    permissionLevel === ENTITY_PERMISSION_LEVEL.TEAM_WRITE ||
-    permissionLevel === ENTITY_PERMISSION_LEVEL.USER_ONLY_WRITE ||
-    permissionLevel === ENTITY_PERMISSION_LEVEL.TEAM_READ_ONLY
-  );
+    return hasEditPermission;
+  } else if (entity.userId === userId) return true;
+
+  return false;
 }
 
-export async function getEntityPermissionLevel(
-  entity: {
-    userId: number | null;
-    teamId: number | null;
-  },
-  userId: number
+export async function canAccessEntity(
+  entity: EntityPermission,
+  userId: number,
+  permission: PermissionString = "routingForm.read"
 ) {
   if (entity.teamId) {
-    const { default: prisma } = await import("@calcom/prisma");
-    const membership = await prisma.membership.findFirst({
-      where: {
-        teamId: entity.teamId,
-        userId,
-        accepted: true,
-      },
+    const permissionService = new PermissionCheckService();
+    const hasReadPermission = await permissionService.checkPermission({
+      teamId: entity.teamId,
+      userId,
+      permission,
+      fallbackRoles: [MembershipRole.ADMIN, MembershipRole.OWNER, MembershipRole.MEMBER],
     });
-    const roleForTeamMember = membership?.role;
 
-    if (roleForTeamMember) {
-      //TODO: Remove type assertion
-      const hasWriteAccessToTeam = (
-        [MembershipRole.ADMIN, MembershipRole.OWNER] as unknown as MembershipRole
-      ).includes(roleForTeamMember);
-      if (hasWriteAccessToTeam) {
-        return ENTITY_PERMISSION_LEVEL.TEAM_WRITE;
-      } else {
-        return ENTITY_PERMISSION_LEVEL.TEAM_READ_ONLY;
-      }
-    }
-  }
-
-  const ownedByUser = entity.userId === userId;
-  if (ownedByUser) {
-    return ENTITY_PERMISSION_LEVEL.USER_ONLY_WRITE;
-  }
+    return hasReadPermission;
+  } else if (entity.userId === userId) return true;
 
-  return ENTITY_PERMISSION_LEVEL.NONE;
-}
-
-async function getMembership(teamId: number | null, userId: number) {
-  const { default: prisma } = await import("@calcom/prisma");
-
-  const team = teamId
-    ? await prisma.team.findFirst({
-        where: {
-          id: teamId,
-          members: {
-            some: {
-              userId,
-              accepted: true,
-            },
-          },
-        },
-        include: {
-          members: {
-            select: {
-              userId: true,
-              role: true,
-            },
-          },
-        },
-      })
-    : null;
-  return team?.members.find((membership) => membership.userId === userId);
+  return false;
 }
 
 export async function canCreateEntity({
   targetTeamId,
   userId,
+  permission = "routingForm.create",
 }: {
   targetTeamId: number | null | undefined;
   userId: number;
+  permission?: PermissionString;
```

Comment:
Why do we need to pass _permission_ argment to a fn that already is used only to check if entity can be created.

I think we can remove permission prop from here and instead directly pass `permission:"routingForm.create"` when calling _checkPermission_